tests/robot-tests/args_and_variables.py

```diff
@@ -82,12 +82,7 @@ def create_argument_parser() -> argparse.ArgumentParser:
         action="store_true",
         help="rerun failed test suites and merge results into original run results",
     )
-    parser.add_argument(
-        "--rerun-attempts",
-        dest="rerun_attempts",
-        type=int,
-        default=0,
-        help="Number of rerun attempts")
+    parser.add_argument("--rerun-attempts", dest="rerun_attempts", type=int, default=0, help="Number of rerun attempts")
```

Comment:
Seems like we've ended up making a tonne of unrelated changes (formatting and logic) across the RF / Python code in this PR. There's _a lot_ of unrelated stuff on top of what is just a simple change to add some guidance notes and is making it a lot longer / harder to read. 

I'm particularly unsure about the Python stuff, particularly as it's also interlaced with formatting changes so I'm just going to have to trust it's correct...

Can we avoid doing this in future PRs? We've previously discussed in retros and dev team meetings to avoid doing stuff like this to keep the scope as narrow as possible.